python/lsst/daf/butler/core/dataUnits.py

```diff
@@ -0,0 +1,39 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Code relating to DataUnits."""
+
+
+class DataUnits:
+    """Represent DataUnits specification.
+
+    Parameters
+    ----------
+    units : `dict`
+        Dictionary of data units keys and values.
+    """
+
+    def __init__(self, units):
+        self.units = units.copy()
+
+    def definedUnits(self):
+        """DataUnits with non-None values."""
+        return {k: v for k, v in self.units.items() if v is not None}
```

Comment:
Not sure why this is needed. Should be absorbed into `DataUnitTypeSet`. But we will have to do that later anyway, so fine to keep it around for now I suppose.

Reply:
I did not know we had a class for that. I asked on Slack but got no hints. Sorry.

Reply:
I can't see a `DataUnitTypeSet` in the codebase.